test/demo/resolveByIntent.inttest.ts

```diff
@@ -1,53 +1,178 @@
+                        const expectedAppNames = getDirectoryAppsForIntent(directory, validIntent);
+
+                        // Resolve the valid intent
+                        const actualAppNames = await findIntent(validIntent).then(apps => apps.map(app => app.name));
+
+                        // Compare the two arrays (sorted as order doesn't matter)
+                        expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                    });
+                });
+            });
+        });
+
+        describe('With an ad-hoc app running', () => {
+            beforeEach(async () => {
+                setupStartNonDirectoryApp(testAppNotInDirectory);
+            });
+
+            describe('But the ad-hoc app has not registered a listener for the intent', () => {
+                test('When calling findIntent with the intent, the ad-hoc app is NOT included in the results', async () => {
+                    const expectedAppNames = getDirectoryAppsForIntent(directory, validIntent);
+
+                    // Resolve the valid intent
+                    const actualAppNames = await findIntent(validIntent).then(apps => apps.map(app => app.name));
+
+                    // Compare the two arrays (sorted as order doesn't matter)
+                    expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                });
+            });
+            describe('And the ad-hoc app has registered a listener for the intent', () => {
+                beforeEach(async () => {
+                    await fdc3Remote.addIntentListener(testAppNotInDirectory, validIntent);
+                    await delay(300);
+                });
+
+                test('When calling findIntent with the intent, the ad-hoc app is returned among the results', async () => {
+                    const directoryAppsForIntent = getDirectoryAppsForIntent(directory, validIntent);
+                    const expectedAppNames = directoryAppsForIntent.concat([testAppNotInDirectory.name]).sort();
+
+                    // Resolve the valid intent
+                    const actualAppNames = await findIntent(validIntent).then(apps => apps.map(app => app.name));
+
+                    // Compare the two arrays (sorted as order doesn't matter)
+                    expect(actualAppNames.sort()).toEqual(expectedAppNames);
+                });
+            });
         });
     });
 
     describe('With context', () => {
         test.todo('TBD what resolving intent with context means as the spec is unclear');
     });
 });
+
+function getDirectoryAppsForIntent(directory: DirectoryApp[], targetIntent: string) {
+    return directory.filter(app => app.intents && app.intents.some(intent => intent.name === targetIntent)).map(app => app.name).sort();
+}
+
+function findIntent(intent: string, context: Context | undefined = undefined, sendFromIdentity?: Identity): Promise<DirectoryApp[]> {
```

Comment:
With this and other similar helpers, how much does the final optional parameter actually get used? It worries me that in cases where it does get used, the parameter order is backwards from `fdc3Remote` conventions, hurting readability of the tests.

I'm cautiously tempted to say if you want to make the call from something other than the testManagerIdentity, just use the fd3Remote, without going through this helper.

I also definitely feel iffy about wrapping a helper around the specific call we're actually testing. Would be interested in a second opinion on this

Reply:
Actually it never gets used :) I agree it's best to remove it and in the rare case a different identity needs to be used, use the `fdc3Remote` form explicitly.

Re. your last concern - Let's see what @pjbroadbent and the rest think. I personally think that such a simple, pure syntactic sugar wrapper doesn't hurt test rigorousness _(is that a word?)_ too much and it makes the code much more readable. Also the signature of the helper is actually closer to the FDC3 API.